### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by enoqueJonas**
# Set up a "Hello world" project as one app with Webpack

## What does this PR do:
- Create a new Rails app called 'hello-rails-react'.
- Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the jsbundling-rails gem is added to your Gemfile.
- Initialize your project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.
- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.

## Improvements to be made:
- None

### CODE REVIEW

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @ 👋,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Status: Invalid Code Review Request



You have submitted a project, that uses the wrong flow. Kindly use [Gitflow.](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/javascript/application.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  }, [])
+  return (<h1>{greetingState.greeting} </h1>);
+}
+
+ReactDOM.render(
+  <Provider store={store}>
+    <Greeting />
+  </Provider>,
+  document.getElementById('root'),
+);
```
Comment (anagudelogu) [inline app/javascript/application.js]: - [ ] You did a good job working on this milestone! 🚀 However, there are some requirements that are not yet met:

image

Please consider creating an `<App />` component and implementing `react-router` with your `<Greeting />` component set as a route. 🤓 🔧

File: app/javascript/redux/greeting/greeting.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// src/reducers/greetingReducer.js
+export const GREETINGS_RETRIEVED = 'hello-rails-react/app/redux/greeting/GREETINGS_RETRIEVED';
+
+const initialState = '';
+
+function greetingReducer(state = initialState, action) {
+    const {type, payload} = action
+    switch (type) {
+        case `${GREETINGS_RETRIEVED}/fulfilled`:
+            console.log('payload')
```
Comment (anagudelogu) [inline app/javascript/redux/greeting/greeting.js]: - [ ] Please don't forget to remove any calls to `console.log` as they kill the performance of the app! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @enoqueJonas, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi @enoqueJonas, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [x] Nice job addressing all the comments from the previous code reviewer :rocket: awesome

### Optional
- [ ] I was not able to run the app with the command `./bin/dev` as required to run the app successfully


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**